app/com/lucidchart/open/cashy/controllers/UploadController.scala

```diff
@@ -137,9 +128,44 @@ class UploadController extends AppController {
     }
   }
 
-  // Returns true if the extension exists (case insensitive)
-  private def checkExtension(key: String): Boolean = {
-    uploadExtensions.exists(e => key.toLowerCase().endsWith(e.toLowerCase()))
+  private def maybeGzip(bytes: Array[Byte], bucket: String, assetName: String, contentType: Option[String]): Boolean = {
```

Comment:
it might be good to put the notion that uploading is going on in the method name so it's evident that that is being done in some cases. maybeGzipAndUpload?